wasm/main.go

```diff
@@ -1,69 +1,674 @@
+									}
+								} else if value == false {
+									switch name {
+									case "LoggedIn":
+										sValue = "LoggedOut"
+									case "DaemonRunning":
+										sValue = "OFFLINE"
+									}
+								}
+								SetDisplay(name, "innerHTML", sValue)
+							}
+							timeStamp := time.Unix(status.TotalUptimePercentage.Timestamp, 0)
+							sTimeStamp := fmt.Sprintf("%s", timeStamp.Format(time.Kitchen))
+							SetDisplay("LastConnected", "innerHTML", sTimeStamp)
+							sFloat := fmt.Sprintf("%.2f", status.TotalUptimePercentage.Percentage)
+							sValue := fmt.Sprintf("%s %s", sFloat, "%")
+							SetDisplay("percentageNumber", "innerHTML", sValue)
+							StartTime = status.SessionStartTime
+						}
+					case "Balance":
+						{
+							log.Debug("Balance Hit")
+							sFloat := fmt.Sprintf("%s", val)
+							for i, value := range sFloat {
+								if strings.ContainsAny(string(value), ".") && (i+5) <= len(sFloat) {
+									sFloat = sFloat[0:i+1] + sFloat[i+1:i+5]
+									break
+								}
+							}
+							sValue := fmt.Sprintf("%s %s", sFloat, "SWRM")
+							log.Debugf("This is Main Balance: %s", sValue)
+							SetDisplay("confirmedBalance", "innerHTML", sValue)
+						}
+					case "Settlement":
+						{
+							log.Debug("Settlement Hit")
+							var settlement Settlement
+							err = json.Unmarshal(val, &settlement)
+							if err != nil {
+								log.Error("Error Unmarshalling settlement: ", err.Error())
+								return
+							}
+							log.Debug("This is Settlement: ", settlement)
+
+							timeZone, err := time.LoadLocation("Local")
+							if err != nil {
+								log.Error("Error while loading Location in Settlement: ", err.Error())
+								return
+							}
+							CurrentZone := (settlement.Date).In(timeZone)
+							date := (CurrentZone).Format("02-01-2006")
+							time := (CurrentZone).Format(time.Kitchen)
+							sDateTime := fmt.Sprintf("%s %s", date, time)
+							SetDisplay("NextDistribution", "innerHTML", sDateTime)
+						}
+					case "BalanceCycle":
+						{
+							log.Debug("BCN Hit")
+							var bcnBalance BCNBalance
+							err = json.Unmarshal(val, &bcnBalance)
+							if err != nil {
+								log.Error("Error in Unmarshalling BCN Balance:", err.Error())
+								return
+							}
+							log.Debug("This is Balance Cycle: ", bcnBalance)
+
+							sValue := fmt.Sprintf("%f %s", (bcnBalance.Owned - bcnBalance.Owe), "SWRM")
+							SetDisplay("Pending", "innerHTML", sValue)
+
+							sDownloaded := fmt.Sprintf("%d %s", (bcnBalance.BytesDownloaded)/1048576, "MB")
+							SetDisplay("CycleDownloaded", "innerHTML", sDownloaded)
+
+							sServed := fmt.Sprintf("%d %s", (bcnBalance.BytesServed)/1048576, "MB")
+							SetDisplay("CycleServed", "innerHTML", sServed)
+						}
+					case "Peers":
+						{
+							log.Debug("Peers Hit")
+							sValue := fmt.Sprintf("%s", val)
+							log.Debugf("This is Number of Peers: %s", val)
+							SetDisplay("PeersData", "innerHTML", sValue)
```

Comment:
I thought you were calling GetPeers from time to time? Instead of doing that, can we call the function to update the peer list display here? We know the list has changed here, so better to do it at this point itself?